ccm_web/server/src/api/api.admin.handler.ts

```diff
@@ -84,18 +84,22 @@ export class AdminApiHandler {
     }
     if (instructor !== undefined) queryParams.by_teachers = ['sis_login_id:' + instructor]
     if (courseName !== undefined) queryParams.search_term = courseName
-    const coursesApiPromises = accountIds.map(async (a) => await this.getAccountCourses(a, queryParams))
+    const coursesApiPromises = createLimitedPromises<CanvasCourse[] | APIErrorData>(
+      accountIds.map(a => async () => await this.getAccountCourses(a, queryParams))
+    )
```

Comment:
We probably don't have to limit one, since we're just making one (albeit potentially rather large paginated request) for each parent account the user has. It can't be that many, even if they're a sub account admin in several accounts. Especially if we're at `20` for a limit, this won't make much of a difference.

Reply:
Good thinking on wrapping the concurrency code to Getting courses from an account, I was bit worried if user search by course name `Bio` then it might fetch lot of courses and will end up with rate limiting errors, this should take care it

Reply:
Just to clarify, this is limiting the number of promises for calls of `this.getAccountCourses`, so if there were three accounts that someone was a sub-account admin for, that would be three promises. The `@kth/canvas-api` library is doing the paging, so that may actually involve multiple calls behind the scenes (probably in sequence?), but for us it's just one promise. Does that make sense? That's why I was saying it was minimal improvement here.

Reply:
Got it.

Reply:
I elected to remove the limiting here. I think it's highly unlikely that anyone would be an admin in >20 sub-accounts. We can reassess if we ever see an edge case scenario.

Reply:
I agree